source/workers/LocalWorker.cpp

```diff
@@ -6297,3 +6537,157 @@ void LocalWorker::anyModeDropCaches()
 			"Path: " + dropCachesPath + "; "
 			"SysErr: " + strerror(errno) );
 }
+
+void LocalWorker::s3ModeGetObjectTags(const std::string& bucketName, const std::string& objectName)
+{
+#ifndef S3_SUPPORT
+    throw WorkerException(std::string(__func__) + "called, but this was built without S3 support");
+#else
+    OPLOG_PRE_OP("GetObjectTagging", bucketName + "/" + objectName, 0, 0);
+
+    const auto& get_tag_outcome = s3Client->GetObjectTagging(
+        S3::GetObjectTaggingRequest()
+            .WithBucket(bucketName)
+            .WithKey(objectName)
+    );
+
+    OPLOG_POST_OP("GetObjectTagging", bucketName + "/" + objectName, 0, 0, !get_tag_outcome.IsSuccess());
+
+    throwOnError(get_tag_outcome, "get tagging", bucketName, objectName);
+
+    // Continue only if we need to verify
+    if (!progArgs->getDoS3ObjectTaggingVerify())
+        return;
+
+    const auto& firstTag = get_tag_outcome.GetResult().GetTagSet().front();
+
+    IF_UNLIKELY(!StringTk::verifyRandomS3TagValue(firstTag.GetValue(), objectName))
+    {
+        std::stringstream errStr;
+        errStr << "Random tag value is corrupted (invalid checksum). "
+               << "Bucket: " << bucketName << "; "
+               << "Key: " << objectName << std::endl
+               << "Tag: " << firstTag.GetKey() << "=" << firstTag.GetValue() << std::endl;
+        throw WorkerException(errStr.str());
+    }
+#endif // S3_SUPPORT
+}
+
+void LocalWorker::s3ModePutObjectTags(const std::string& bucketName, const std::string& objectName)
+{
+#ifndef S3_SUPPORT
+    throw WorkerException(std::string(__func__) + "called, but this was built without S3 support");
+#else
+    const auto tag = S3::Tag()
+        .WithKey(TAG_KEY_MEDIUM_NAME)
+        .WithValue(StringTk::generateRandomS3TagValue(objectName, TAG_VALUE_MEDIUM_LEN));
+
+    OPLOG_PRE_OP("PutObjectTagging", bucketName + "/" + objectName, 0, TAG_VALUE_MEDIUM_LEN);
+
+    const auto& put_tag_outcome = s3Client->PutObjectTagging(
+        S3::PutObjectTaggingRequest()
+            .WithBucket(bucketName)
+            .WithKey(objectName)
+            .WithTagging(S3::Tagging().AddTagSet(tag))
+    );
+
+    OPLOG_POST_OP("PutObjectTagging", bucketName + "/" + objectName,
+                  0, TAG_VALUE_MEDIUM_LEN, !put_tag_outcome.IsSuccess());
+
+    throwOnError(put_tag_outcome, "put tagging", bucketName, objectName);
```

Comment:
variable names are camelCase. (this also applies to other places in this patch, e.g. `del_tag_outcome`)